aten/src/ATen/native/GridSampler.cpp

```diff
@@ -380,12 +381,278 @@ namespace {
     return std::make_tuple(grad_input, grad_grid);
   }
 
+  template<typename scalar_t>
+  Tensor grid_sampler_2d_cpu_impl(const Tensor& input, const Tensor& grid,
```

Comment:
I'm still trying to understand the problem with the vectorized instructions. I'm not familiar with AVX, but is it possible to template on the index type in the vectorized implementation? The downside of maintaining a separate implementation here for 64-bit indexing is that anyone who updates the operation in the future will need to make changes to both implementations.

Reply:
As I mentioned in the PR description, there are `i64_gather_ps` instructions that uses 64-bit indexing for floats. However, that only loads in half of the AVX register so makes things more complicated. It can be done, but: the code will be harder to read, the performance won't benefit much and again there is a trade of in binary size. Do we want to compile 3 vector instructions * 2 interpolation modes * 3 padding types * 2 align_corners modes = 36 versions of the function when it's likely to be memory bound for such large tensors anyway?

Reply:
Let's just stick with the fallback for now. If someone wants to vectorize the fallback path for 64-bit indexing and float inputs then they can go do that but I doubt we'll get a feature request for that anytime soon.